tests/autobahn/client.py

```diff
@@ -6,31 +6,32 @@
 
 
 async def client(loop, url, name):
-    ws = await aiohttp.ws_connect(url + "/getCaseCount")
-    num_tests = int((await ws.receive()).data)
-    print("running %d cases" % num_tests)
-    await ws.close()
-
-    for i in range(1, num_tests + 1):
-        print("running test case:", i)
-        text_url = url + "/runCase?case=%d&agent=%s" % (i, name)
-        ws = await aiohttp.ws_connect(text_url)
-        while True:
-            msg = await ws.receive()
-
-            if msg.type == aiohttp.WSMsgType.TEXT:
-                await ws.send_str(msg.data)
-            elif msg.type == aiohttp.WSMsgType.BINARY:
-                await ws.send_bytes(msg.data)
-            elif msg.type == aiohttp.WSMsgType.CLOSE:
-                await ws.close()
-                break
-            else:
-                break
-
-    url = url + "/updateReports?agent=%s" % name
-    ws = await aiohttp.ws_connect(url)
-    await ws.close()
+    async with aiohttp.ClientSession() as session:
+        async with session.ws_connect(url + "/getCaseCount") as ws:
+            num_tests = int((await ws.receive()).data)
+            print("running %d cases" % num_tests)
+            await ws.close()
+
+            for i in range(1, num_tests + 1):
+                print("running test case:", i)
+                text_url = url + "/runCase?case=%d&agent=%s" % (i, name)
+                ws = await session.ws_connect(text_url)
+                while True:
```

Comment:
This `while` loop can be replaced with the  `__aiter__` call on the `ws` object. The `CLOSE` message will be handled by the library :) 

```python3
async for msg in ws:
    if msg.type == aiohttp.WSMsgType.TEXT:
        await ws.send_str(msg.data)
    elif msg.type == aiohttp.WSMsgType.BINARY:
        await ws.send_bytes(msg.data)
    else:
        break
```

Reply:
That is much cleaner, thank you